Carbon.Redis/Abstractions/ICarbonCacheExtensions.cs

```diff
@@ -211,16 +211,15 @@ public static async Task<(ICarbonCache, string)> HashSingleSet<T>(this ICarbonCa
         public static async Task<T> HashSingleGet<T>(this ICarbonCache instance, string key) where T : class
         {
             var value = await instance.GetDatabase().HashGetAsync(key.ToRedisInstanceKey(instance), HashData, CommandFlags.PreferReplica);
-            T result;
-
-            try
+            T result = ((byte[])value).FromByteArray<T>(SerializationType);
+            if (result is null)
             {
-                result = ((byte[])value).FromByteArray<T>(SerializationType);
-            }
-            catch (JsonException)
-            {
-                result = ((byte[])value).FromByteArray<T>(CarbonContentSerializationType.BinaryFormatter);
-                await instance.HashSingleSet<T>(key, result);
+                result = TryAlternativeDeserialization<T>((byte[])value);
+                if (result != null)
+                {
+                    HashEntry hashEntry = new HashEntry(HashData, result.ToByteArray(CarbonContentSerializationType.Json));
```

Comment:
(opsiyonel) buraya ilerisi için bir comment bırakabiliriz belki, json geçiş süreci tamamlandıktan sonra kaldırılacaktır diye